sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -587,13 +587,26 @@ fromClause
     ;
 
 aggregationClause
-    : GROUP BY groupingExpressions+=expression (',' groupingExpressions+=expression)* (
+    : GROUP BY groupingExpressionWithGroupingAnalytics+=groupByClause
+      (',' groupingExpressionWithGroupingAnalytics+=groupByClause)*
+    | GROUP BY groupingExpressions+=expression (',' groupingExpressions+=expression)* (
       WITH kind=ROLLUP
     | WITH kind=CUBE
     | kind=GROUPING SETS '(' groupingSet (',' groupingSet)* ')')?
     | GROUP BY kind=GROUPING SETS '(' groupingSet (',' groupingSet)* ')'
     ;
 
+groupByClause
+    : groupingAnalytics
+    | expression
+    ;
+
+groupingAnalytics
+    : ROLLUP  '(' expression (',' expression)* ')'
+    | CUBE '(' expression (',' expression)* ')'
+    | GROUPING SETS '(' groupingSet (',' groupingSet)* ')'
```

Comment:
`kind=(ROLLUP | CUBE | GROUPING SETS) '(' groupingSet (',' groupingSet)* ')'`?

Reply:
> `kind=(ROLLUP | CUBE | GROUPING SETS) '(' groupingSet (',' groupingSet)* ')'`?

You reminded me that cube and rollup are basically the same Grouping， need a big change.

Reply:
Ah, bad. I meant `kind=(ROLLUP | CUBE) '(' groupingSet (',' groupingSet)* ')'?`

Reply:
> Ah, my bad. I meant `kind=(ROLLUP | CUBE) '(' expressions (',' expressions)* ')'?`

I checked with PostgresSQL... it really support  `kind=(ROLLUP | CUBE | GROUPING SETS) '(' groupingSet (',' groupingSet)* ')'?`

Reply:
Could you file a jira for it and check if the other systems (e.g., Oracle, MySQL, ...) support the syntax?

Reply:
> Could you file a jira for it and check if the other systems (e.g., Oracle, MySQL, ...) support the syntax?

https://issues.apache.org/jira/browse/SPARK-33377
How about:

1.  Just remove `cube|rollup` function and move `cube|rollup|groupings sets` to parser level and don't support composite column in this pr.
2. A new pr of  https://issues.apache.org/jira/browse/SPARK-33377 to support composite column in cube & rollup
3. Pr to replace current `GroupingSets`  https://issues.apache.org/jira/browse/SPARK-33309
3. PR support  partial `cube|rollup|grouping` set and all mixed case  https://github.com/apache/spark/pull/30144
4 Then pr support CUBE/ROLLUP/GROUPING SETS use GROUP BY ordinal  https://github.com/apache/spark/pull/30145